### PR TITLE
Refactor properties

### PR DESCRIPTION
**Pull request by Shutgun**
## Proposed change

Refactor properties to not need a complete connection set but a setter method that is called instead.

## Checklist

- [ ] Version number in \_\_init\_\_.py was properly set.
- [ ] Changelog is updated.

### CODE REVIEW

File: devolo_home_control_api/backend/mprm_rest.py [hunk 1]
```diff
@@ -71,11 +71,95 @@ def get_name_and_element_uids(self, uid: str):
...
+        Set a binary switch state of a device.
+
+        :param uid: Element UID, something like devolo.BinarySwitch:hdm:ZWave:CBC56091/24
+        :param state: True if switching on, False if switching off
+        :return: True if successfully switched, false otherwise
+        """
+        data = {"method": "FIM/invokeOperation",
+                "params": [uid, "turnOn" if state else "turnOff", []]}
+        response = self._post(data)
+        if response['result']['status'] == 1:
```
Comment (2Fake) [inline devolo_home_control_api/backend/mprm_rest.py:98]: Checking the response status can be wrapped into a function as it looks the same everytime (almost)

File: devolo_home_control_api/properties/multi_level_sensor_property.py [hunk 2]
```diff
@@ -48,10 +41,9 @@ def unit(self, unit: int):
                  "voltage": {0: "V", 1: "mV"}
                  }
         try:
-            self._unit = units[self.sensor_type].get(unit, str(unit))
+            return units[self.sensor_type].get(self._unit, str(self._unit))
```
Comment (2Fake) [inline devolo_home_control_api/properties/multi_level_sensor_property.py:44]: Can be simplified like this:

```suggestion
            return units.get(self.sensor_type).get(self._unit, str(self._unit)) or str(self.unit)
```

File: devolo_home_control_api/properties/settings_property.py [hunk 3]
```diff
@@ -10,22 +10,24 @@ class SettingsProperty(Property):
...
 
-        super().__init__(connection=connection, element_uid=element_uid)
-        for key, value in kwargs.items():
-            setattr(self, key, value)
+        super().__init__(element_uid=element_uid)
+        self._setter = setter
 
         if element_uid.startswith("gds"):
-            self.zone = self._gateway.zones[self.zone_id]
+            self.zone = kwargs.pop("zones")[kwargs["zone_id"]]
```
Comment (2Fake) [inline devolo_home_control_api/properties/settings_property.py:27]: Is a KeyError possible here?

File: devolo_home_control_api/properties/multi_level_sensor_property.py (hunk 2 above)
Comment (Shutgun) [reply devolo_home_control_api/properties/multi_level_sensor_property.py:44]: No, that will not work as, we try to access get on a NoneType.

File: devolo_home_control_api/backend/mprm_rest.py (hunk 1 above)
Comment (Shutgun) [reply devolo_home_control_api/backend/mprm_rest.py]: Done

File: devolo_home_control_api/properties/settings_property.py (hunk 3 above)
Comment (Shutgun) [reply devolo_home_control_api/properties/settings_property.py:27]: Yes it is. I took care of that.

Comment (2Fake) [reply devolo_home_control_api/properties/settings_property.py:27]: Can be simplified like this:

`if element_uid.startswith("gds") and all(e in kwargs for e in ["zones", "zone_id"]):`

Comment (Shutgun) [reply devolo_home_control_api/properties/settings_property.py:27]: "Done"

**Pull request merged**